src/compare.py

```diff
@@ -126,6 +133,59 @@ def compare_phash(source: MatLike, capture: MatLike, mask: MatLike | None = None
     return 1 - (hash_diff / 64.0)
 
 
+# copied from https://github.com/madmaze/pytesseract
+def subprocess_args():
+    # See https://github.com/pyinstaller/pyinstaller/wiki/Recipe-subprocess
+    # for reference and comments.
+
+    kwargs = {
+        "stdin": subprocess.PIPE,
+        "stdout": subprocess.PIPE,
+        "stderr": subprocess.DEVNULL,
+        "startupinfo": None,
+        "env": environ,
+    }
+
+    if hasattr(subprocess, "STARTUPINFO"):
+        kwargs["startupinfo"] = subprocess.STARTUPINFO()
+        kwargs["startupinfo"].dwFlags |= subprocess.STARTF_USESHOWWINDOW
+        kwargs["startupinfo"].wShowWindow = subprocess.SW_HIDE
+
+    return kwargs
+
+
+def run_tesseract(capture: MatLike):
+    png = np.array(cv2.imencode(".png", capture)[1]).tobytes()
+    p = subprocess.Popen(TESSERACT_CMD, **subprocess_args())
+    output = p.communicate(input=png)[0]
+    return output.decode(DEFAULT_ENCODING)
+
+
+def extract_and_compare_text(capture: MatLike, texts: list[str]):
+    """
+    Compares the extracted text of the given image and returns the similarity between the two texts.
+    The best match of all texts is returned.
+
+    @param capture: Image of any given shape as a numpy array
+    @param texts: a list of strings to match for
+    @return: The similarity between the text in the image and the text supplied as a number 0 to 1.
+    """
+    # if the string is found 1:1 in the string extracted from the image a 1 is returned.
+    # otherwise the levenshtein ratio is calculated between the two strings and gets returned.
+    image_string = run_tesseract(capture).lower().strip()
+
+    ratio = 0.0
+    for text in texts:
+        if text in image_string:
```

Comment:
This is arguable. Should, for example, `"Goal"` really match `"Here's the goalkeeper"` ? Given that we expect a fixed position and size, and we support multiple text inputs for a split?

Reply:
I wanted to provide a fast perfect match option, which was simpel and not dependend on fuzzy logic.

In fact I was using this option for different split outcomes but all of them containing the same word.

When we speak about a game someone would need to visit every split outcome first and note the exact texts for all possible versions.

But if the player knows a word that is contained in every one of them, this option saves time in that regard.

Reply:
I'm not convinced this isn't gonna lead to false-positives issues. But if you can at least explain the reasoning as a comment in code, or link this your comment here on GitHub.

Then we can always revisit later.